packages/test/test-end-to-end-tests/src/test/gc/gcDeleteObjectsInTestMode.spec.ts

```diff
@@ -195,6 +195,10 @@ describeFullCompat("GC delete objects in test mode", (getTestObjectProvider) =>
             const container = await provider.makeTestContainer(testContainerConfig);
             mainDataStore = await requestFluidObject<ITestDataObject>(container, "/");
             containerRuntime = mainDataStore._context.containerRuntime as ContainerRuntime;
+
+            // Send an op before GC runs. GC needs current timestamp to work with which is retrieved from ops. Without
+            // any op, GC will not run.
+            mainDataStore._root.set("key", "value");
```

Comment:
Why doesn't the join op get generated after all this happens?

Reply:
The join op will be generated when this op is sent.

Reply:
Oh, that's interesting. We only generate the join op when we write the first op

Reply:
The container joins in read mode by default and there is no join op in read mode. The first op is nack'd resulting in reconnection in write mode which results in a join op.

Reply:
This is quite an interesting flow, I would expect write mode clients to join in write mode immediately.

Reply:
The default mode is read. If a client wants to connect in write mode by default, it has to explicitly specify that via loaded params.

Reply:
@tyler-cai-microsoft I believe I was wrong about the default mode. I believe the default mode for the first container (detached) is read but for subsequent clients, the mode is write. There were discussions about making the mode write for all containers but looks like it never happened. See this comment - https://github.com/microsoft/FluidFramework/blob/076af15a8ea07ded5e10bb25aef3e64d9e759907/packages/loader/container-loader/src/container.ts#L1129